xbmc/platform/win32/DisplayUtilsWin32.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DisplayUtilsWin32.h"
+
+#include "ServiceBroker.h"
+#include "utils/StringUtils.h"
+#include "utils/SystemInfo.h"
+#include "utils/log.h"
+
+extern HWND g_hWnd;
+
+std::wstring CDisplayUtilsWin32::GetCurrentDisplayName()
+{
+  MONITORINFOEXW mi{};
+  mi.cbSize = sizeof(mi);
+  if (!GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi))
+    return {};
+  return mi.szDevice;
+}
+
+std::vector<DISPLAYCONFIG_PATH_INFO> CDisplayUtilsWin32::GetDisplayConfigPaths()
+{
+  uint32_t pathCount{0};
+  uint32_t modeCount{0};
+  std::vector<DISPLAYCONFIG_PATH_INFO> paths;
+  std::vector<DISPLAYCONFIG_MODE_INFO> modes;
+
+  constexpr uint32_t flags{QDC_ONLY_ACTIVE_PATHS};
+  LONG result{ERROR_SUCCESS};
+
+  do
+  {
+    if (GetDisplayConfigBufferSizes(flags, &pathCount, &modeCount) != ERROR_SUCCESS)
+      return {};
+
+    paths.resize(pathCount);
+    modes.resize(modeCount);
+
+    result = QueryDisplayConfig(flags, &pathCount, paths.data(), &modeCount, modes.data(), nullptr);
+  } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+  if (result != ERROR_SUCCESS)
+    return {};
+
+  paths.resize(pathCount);
+  modes.resize(modeCount);
+
+  return paths;
+}
+
+std::optional<CDisplayUtilsWin32::DisplayConfigId> CDisplayUtilsWin32::GetCurrentDisplayTargetId()
+{
+  std::wstring name{GetCurrentDisplayName()};
+  if (name.empty())
+    return std::nullopt;
+
+  return GetDisplayTargetId(name);
+}
+
+std::optional<CDisplayUtilsWin32::DisplayConfigId> CDisplayUtilsWin32::GetDisplayTargetId(
+    const std::wstring& gdiDeviceName)
+{
+  DISPLAYCONFIG_SOURCE_DEVICE_NAME source{};
+  source.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+  source.header.size = sizeof(source);
+
+  for (const auto& path : GetDisplayConfigPaths())
+  {
+    source.header.adapterId = path.sourceInfo.adapterId;
+    source.header.id = path.sourceInfo.id;
+
+    if (DisplayConfigGetDeviceInfo(&source.header) == ERROR_SUCCESS &&
+        gdiDeviceName == source.viewGdiDeviceName)
+      return DisplayConfigId{path.targetInfo.adapterId, path.targetInfo.id};
+  }
+  return std::nullopt;
+}
+
+HDR_STATUS CDisplayUtilsWin32::SetDisplayHDRStatus(const DisplayConfigId& identifier, bool enable)
+{
+  LONG result{ERROR_SUCCESS};
+  HDR_STATUS status{enable ? HDR_STATUS::HDR_ON : HDR_STATUS::HDR_OFF};
+  CLog::LogF(LOGINFO, "Set Windows HDR {}.", enable ? "On" : "Off");
```

Comment:
For consistency maybe this log line can be moved to CWIN32Util just before call `CDisplayUtilsWin32::SetDisplayHDRStatus` is the same and not need `#include log.h` here

Reply:
sure.